src/app/utility/UrlUtil.ts

```diff
@@ -14,10 +14,14 @@ import * as m from 'mithril';
  * Set the query parameters in the URL to be equal to the key value pairs in filters.
  * @param filters
  */
-export const setQueryParams = (filters: FilterState): void => {
+export const setQueryParams = (filters: FilterState, isModal: boolean): void | string => {
```

Comment:
Remove isModal and remove the string return type.
Then create a new function that accepts filters and returns a query string for truthyFilters (i.e. everything except the m.route.set call), then use your newly created function in the modal and call the function inside setQueryParams